gravitee-am-service/src/main/java/io/gravitee/am/service/impl/user/BaseUserEnhancer.java

```diff
@@ -37,9 +37,16 @@ public abstract class BaseUserEnhancer implements UserEnhancer {
     protected final Logger LOGGER = LoggerFactory.getLogger(getClass());
     protected abstract Flowable<Group> getGroupsByMemberId(String memberId);
     protected abstract Flowable<Role> getRolesByIds(List<String> roleIds);
+    protected abstract Flowable<Group> getGroupsByIds(List<String> groupIds);
+
+    private Flowable<Group> collectGroups(User user) {
+        Flowable<Group> groupsByMemberId = getGroupsByMemberId(user.getId());
+        Flowable<Group> groupsByIds = getGroupsByIds(user.getDynamicGroups());
+        return groupsByMemberId.mergeWith(groupsByIds);
```

Comment:
question: should we ensure somehow uniquess in the final Flowable as merged flows may have duplicates ?

Reply:
good point. I convert List to Set in the end.